ably.ts.d

```diff
@@ -0,0 +1,457 @@
+    capability: String;
+    clientId?: String;
+    nonce?: String;
+    timestamp: Number;   // Time?
+    ttl: Number;
+  }
+
+  interface CipherParams {
+    algorithm: String;
+    key: any;
+    keyLength: number;
+    mode: String;
+  }
+
+  interface ErrorInfo {
+    code: number;
+    message: String;
+    statusCode: number;
+  }
+
+  interface ConnectionDetails {
+    clientId: String;
+    connectionKey: String;
+    connectionStateTtl: number;
+    maxFrameSize: number;
+    maxInboundRate: number;
+    maxMessageSize: number;
+    serverId: String;
+    maxIdleInterval: number;
+  }
+
+  interface StatsMessageTypes {
+    all: number,
+    messages: number,
+    presence: number;
+  }
+
+  interface StatsRequestCount {
+    failed: number,
+    refused: number,
+    succeeded: number
+  }
+
+  interface StatsResourceCount {
+    mean: number,
+    min: number,
+    opened: number,
+    peak: number,
+    refused: number
+  }
+
+  interface StatsConnectionTypes {
+    all: number,
+    plain: number,
+    tls: number
+  }
+
+  interface StatsMessageTraffic {
+    all: number,
+    realtime: number,
+    rest: number,
+    webhook: number
+  }
+
+  // Listeners
+  type ListenerFn = (...args: Array<any>) => void;
+
+
+  // Internal Classes
+  declare class EventEmitter {
+    on: (eventOrCallback: String | ListenerFn, callback?: ListenerFn) => void;
+
+    once: (eventOrCallback: String | ListenerFn, callback?: ListenerFn) => void;
+
+    off: (eventOrCallback?: String | ListenerFn, callback?: ListenerFn) => void;
+
+    emit: (Event: String, ...args: Array<any>) => void;
+  }
+
+  // Classes
+  export declare class Auth {
+    clientId: String;
+    authorize: (tokenParams?: TokenParams, authOptions?: AuthOptions) => TokenDetails;
+    createTokenRequest: (tokenParams?: TokenParams, authOptions?: AuthOptions) => TokenRequest;
+    requestToken: (TokenParams?: TokenParams, authOptions?: AuthOptions) => TokenDetails;
+  }
+
+  export declare class TokenDetails {
+    fromJson: (JsonData: string) => TokenDetails;
+    capability: String;
+    clientId: String;
+    expires: number; // Time
+    issued: number; // Time
+    token: String;
+  }
+
+  export declare class TokenRequest {
+    fromJson: (JsonData: String) => TokenRequest;
+    capability: String;
+    clientId: String;
+    keyName: String;
+    mac: String;
+    nonce: String;
+    timestamp: number; // Time
+    ttl: number;
+  }
+
+  export declare class ChannelOptions {
+    withCipherKey: (key: any) => ChannelOptions;
+    cipher: (params: CipherParams) => void;
+  }
+
+  export declare class RestPresence {
+    get: (limit: number, clientId?: String, connectionId?: String) => PaginatedResult;
+
+    history: (start: number, end: number, direction: String, limit: number) => PaginatedResult;
+  }
+
+  export declare class RealtimePresence {
+    syncComplete: boolean;
+
+    get: (waitForSync: boolean, clientId?: String, connectionId?: String) => PresenceMessage;
```

Comment:
This takes a params object, not three separate parameters. (similarly for lots of other fns here)